File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
class Solution {
    public int[] solution(String str) {
        int length = str.length();
        int[] arr = new int[length];
        arr[0] = -1;
        
        for(int i = 1; i < length; i++) {
            for(int j = 0; j < i; j++) {
                if(str.charAt(i) == str.charAt(j)) {
                    arr[i] = i-j;
                } else {
                    if(arr[i] == 0) {
                        arr[i] = -1;
                    }
                }
            }
        }
        return arr;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
const solution = (s) => {
    const map = new Map();
    
    const result = [...s].map((char, i) => {
        if(map.has(char)) {
            const diff = i - map.get(char);
            map.set(char, i);
            return diff;
        } else {
            map.set(char, i);
            return -1;
        }
    })
    
    return result;
}